### PR TITLE
add astester.imports.js

### PR DESCRIPTION
**Pull request by SebastienValla**
add a envy.imports.js to mock VM and make unit tests with for the moment : 
generateEvent
setOf
getOf

### CODE REVIEW

File: astester.imports.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  return {
+    env: {
+      memory,
+      abort(msg, file, line, col) {
+      getString(msg)
+
+       console.log("error", text)
+      },
+      log(ptr) {
+       console.log("log", text)
```
Comment (gregLibert) [inline astester.imports.js:12]: How can that work ? What is this magic ?

File: astester.imports.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+      },
+  },
+  massa: {
+      memory,
```
Comment (gregLibert) [inline astester.imports.js:16]: not sure you need that.

File: astester.imports.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+      assembly_script_generate_event(string) {
+       console.log("event", string)
```
Comment (gregLibert) [inline astester.imports.js:18]: not sure you can do that. I think you will have a pointer. You should have a look at what Max did for the playground.

File: astester.imports.js [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+      },
+      assembly_script_set_data_for(address, key, value) {
```
Comment (gregLibert) [inline astester.imports.js:20]: Same, if it's string, you will have pointer

File: astester.imports.js [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+      },
+      assembly_script_get_data_for(address, key) {
+          let value = "";
+          if (Storage.has(address)) {
+              const addressStorage = Storage.get(address);
+              if (addressStorage.has(key)) {
+                  value = addressStorage.get(key);
+              }
+          }
+          return value;
```
Comment (gregLibert) [inline astester.imports.js:35]: Same, not sure it will work. You should return a pointer from distant memory after setting the string there.

**Review by gregLibert — CHANGES_REQUESTED** [review]

File: astester.imports.js (hunk 1 above)
Comment (SebastienValla) [reply astester.imports.js]: yes indeed it was ! issue solve with getString

File: astester.imports.js (hunk 2 above)
Comment (SebastienValla) [reply astester.imports.js:31]: needed to include it in astester/cli

File: astester.imports.js (hunk 3 above)
Comment (SebastienValla) [reply astester.imports.js]: indeed, problem solved

**Review by gregLibert — APPROVED** [review]

**Review by peterjah — APPROVED** [review]

**Pull request merged**